relay-server/src/utils/envelope_context.rs

```diff
@@ -89,82 +117,39 @@ impl EnvelopeContext {
                 self.summary.profile_quantity,
             );
         }
-    }
 
-    /// Records an outcome scoped to this envelope's context.
-    pub fn track_outcome(&self, outcome: Outcome, category: DataCategory, quantity: usize) {
-        let outcome_aggregator = OutcomeAggregator::from_registry();
-        outcome_aggregator.do_send(TrackOutcome {
-            timestamp: self.received_at,
-            scoping: self.scoping,
-            outcome,
-            event_id: self.event_id,
-            remote_addr: self.remote_addr,
-            category,
-            // Quantities are usually `usize` which lets us go all the way to 64-bit on our
-            // machines, but the protocol and data store can only do 32-bit.
-            quantity: quantity as u32,
-        });
+        self.finish(RelayCounters::EnvelopeRejected);
     }
 
     /// Returns scoping stored in this context.
     pub fn scoping(&self) -> Scoping {
         self.scoping
     }
 
-    /// Returns the event id of this context, if any.
-    pub fn event_id(&self) -> Option<EventId> {
-        self.event_id
+    /// Returns the instant at which the envelope was received at this Relay.
+    ///
+    /// This is the monotonic time equivalent to [`received_at`](Self::received_at).
+    pub fn start_time(&self) -> Instant {
```

Comment:
naming is hard, but maybe hungarian naming is defensible in this case: `.received_instant()`?  currently these two names are just so far apart.

Reply:
Agreed. So far the entire code base consistently calls them `start_time` and `received_at` because of historical context. If it's OK, I'd do a sweep in a dedicated PR for this.